src/System.IO.Pipelines.Extensions/UnownedBufferReader.cs

```diff
@@ -243,6 +243,11 @@ void IPipeReader.Complete(Exception exception)
             }
         }
 
+        public void OnWriterCompleted(Action<Exception, object> callback, object state)
+        {
+            throw new NotImplementedException();
```

Comment:
This PR is big enough, and I don't want to write tests for `UnownedBufferReader`. Lets decide if we want to keep it at all.